xarray/tests/test_formatting.py

```diff
@@ -555,6 +556,114 @@ def test_array_scalar_format(self) -> None:
             format(var, ".2f")
         assert "Using format_spec is only supported" in str(excinfo.value)
 
+    def test_datatree_print_empty_node(self):
+        dt: DataTree = DataTree(name="root")
+        printout = dt.__str__()
+        assert printout == "DataTree('root', parent=None)"
+
+    def test_datatree_print_empty_node_with_attrs(self):
+        dat = xr.Dataset(attrs={"note": "has attrs"})
+        dt: DataTree = DataTree(name="root", data=dat)
+        printout = dt.__str__()
+        assert printout == dedent(
+            """\
+            DataTree('root', parent=None)
+                Dimensions:  ()
+                Data variables:
+                    *empty*
+                Attributes:
+                    note:     has attrs"""
+        )
+
+    def test_datatree_print_node_with_data(self):
+        dat = xr.Dataset({"a": [0, 2]})
+        dt: DataTree = DataTree(name="root", data=dat)
+        printout = dt.__str__()
+        expected = [
+            "DataTree('root', parent=None)",
+            "Dimensions",
+            "Coordinates",
+            "a",
+            "Data variables",
+            "*empty*",
+        ]
+        for expected_line, printed_line in zip(expected, printout.splitlines()):
+            assert expected_line in printed_line
+
+    def test_datatree_printout_nested_node(self):
+        dat = xr.Dataset({"a": [0, 2]})
+        root: DataTree = DataTree(name="root")
+        DataTree(name="results", data=dat, parent=root)
+        printout = root.__str__()
+        assert printout.splitlines()[2].startswith("    ")
+
+    def test_print_datatree(self, simple_datatree):
+        dt = simple_datatree
+        print(dt)
+
+        # TODO work out how to test something complex like this
```

Comment:
This was a placeholder for something better. All this does right now is check an error is raised. But we might not need more tests than the ones already in this PR.

Reply:
Let's just pull this test, as we are already testing that the `__str__` method doesn't error in other tests.